cmd/controller-manager/app/options/options.go

```diff
@@ -143,6 +143,17 @@ type Options struct {
 	// in scenario of dynamic replica assignment based on cluster free resources.
 	// Disable if it does not fit your cases for better performance.
 	EnableClusterResourceModeling bool
+
+	// DisableMultiDependencyDistribution indicates disable the ability to a resource from being depended on by multiple
+	// resources or being distributed by PropagationPolicy/ClusterPropagationPolicy while being depended on.
+	//
+	// Before v1.12, this capability is allowed by default. If you still wish to enable this capability, you can set
+	// this flag to false. However, you will need to bear some side effects that come with it.
+	// For example, you can refer to https://github.com/karmada-io/karmada/pull/5717. When the primary resource is deleted,
+	// it does not consider other resources that currently depend on the resource or any PropagationPolicy associated with it.
+	//
+	// It is recommended that you adapt your business accordingly to avoid continued use.
+	DisableMultiDependencyDistribution bool
```

Comment:
Would it be better to have a feature gate than a command flag? Something like:  `DisableDependencySharing`.
For example:
- release-1.12 starts from alpha, which is disabled by default, for backward compatibility concerns.
- release-1.23 move to beta, which is enabled by default, and allows people to disable it as an escape hatch
- release-1.24 move to stable, enabled by default, and don't allow to disable it.

@XiShanYongYe-Chang @chaunceyjiang What do you think?